### ISSUE TITLE
useLocation somehow breaks this component

### ISSUE THREAD

**Issue by Unforgiven-wanda** [DevbookHQ/splitter#2]
Hello, and thank you for this component.
I am using this component inside a Tabpane (provided by Antd) and so far it works fine, but somehow, adding React-router-dom's `useLocation` hook throws an error when switching tabs.

- On enabling the tab containing the Splitter and useLocation the component displays fine
- Switching to any other tab however throws `Cannot create pairs - parent has undefined or zero size: undefined.` error

Simply remove useLocation from the component and it starts working perfectly again.
I honestly haven't the faintest idea why it behaves that way, I've been at it all morning and I'm all out of ideas atm.
Any pointers would be most helpful.

Thank you.

**Comment by mlejva** [comment]
If you could share an example CodeSandbox project (or similar tool) it would help a lot to debug this.

**Comment by mlejva** [comment]
I think the problem is that the parent's size isn't [correctly calculated](https://github.com/DevbookHQ/splitter/blob/master/src/index.tsx#L156). I'm not exactly sure why it happens with `useLocation` though. I can say more once I see an example.

**Comment by Unforgiven-wanda** [comment]
Hello again,
Unfortunately I haven't been able to set up a simple example. The project is long since unmanageable. 
I did start setting up a minimal example on [StackBlitz](https://stackblitz.com/edit/react-ts-hlbyxb?file=components/index.ts) but am ashamed to admit I gave up halfway as it became clear I wouldn't be able to reproduce it. Still the link should give you a broad idea of my app structure.
Barring that I have made a [small video](https://files.catbox.moe/pcpcme.mp4) demonstrating what's happening, hopefully it helps.
Again any and all pointers would be most welcome, I'm not so much looking for a fix as I'm looking for clues as to what is happening.

Thank you.
